cdap-docs/deploy.sh

```diff
@@ -0,0 +1,201 @@
+# Deployment script for docs
+# Deploys zip files created by build scripts
+# 
+# BUILD_WORKING_DIR must be passed as an environment variable (can be set in Bamboo)
+#   It will typically look like this:
+#   Release: /var/bamboo/xml-data/build-dir/CDAP-DRBD-JOB1
+#   Develop: /var/bamboo/xml-data/build-dir/CDAP-DDBD-JOB1
+# 
+# OPT_DIR is an optional directory to use instead of derived branch name. E.g.:
+#   if branch name=feature/docs-build-testing, and we want to use testdir
+#   we would set OPT_DIR=testdir and add it to the bamboo tasks' variables
+#   and the remote directory would be 2.8.1/testdir (testdir treated as branch)
+# If OPT_DIR is not set, bamboo.planRepository.<position>.branch will be used,
+#   e.g. /var/www/html/cdap/2.8.1/feature-myfix, unless the branch name
+#   is release/* or develop/* in which case, it will just be put into
+#   the version directory, e.g. /var/www/html/cdap/2.8.1
+#
+################################################################################
+
+# output trimmer
+decho () {
+  if [[ "${DEBUG}" == 'yes' ]]; then
+    echo ${*}
+  else
+    RSYNC_QUIET='--quiet'
+  fi
+}
+
+# set the mandatory variables first (from info in project's main pom.xml)
+
+# get project version from main pom.xml
+function get_version () {
+  TMP_VERSION=`grep "<version>" ../pom.xml`
+  TMP_VERSION=${TMP_VERSION#*<version>}
+  VERSION=${TMP_VERSION%%</version>*}
+}
+
+# get project name (e.g. cdap) from main pom.xml
+function get_project () {
+  TMP_PROJECT=`grep "<artifactId>" ../pom.xml`
+  TMP_PROJECT=${TMP_PROJECT#*<artifactId>}
+  PROJECT=${TMP_PROJECT%%</artifactId>*}
+}
+
+# convert branch names that look like this: feature/my-branch to feature-my-branch
+function convert_branch_name () {
+  decho "converting '/' to '-' in branch name"
+  DOC_DIR=`echo $DOC_DIR | tr '/' '-'`
+}
+
+# create remote directory based on type of build
+#   3 scenarios (assuming remote base web directory=/var/www/html/cdap and VERSION=2.8.1)
+#     OPT_DIR is set via environment variables
+#         => remote directory=/var/www/html/cdap/${VERSION}-${OPT_DIR}
+#     OPT_DIR is not set AND branch=release/* or branch=develop/*
+#         => remote directory=/var/www/html/cdap/2.8.1
+#     OPT_DIR is not set AND branch=anything else (e.g. feature/*, hotfix/*, etc..)
+#         => remote directory=/var/www/html/cdap/${VERSION}-${BRANCH_NAME}
+set_remote_dir () {
+  DOC_DIR=${OPT_DIR:-${BRANCH_NAME}}
+  convert_branch_name
+  BRANCH=''
+  if [[ "${DOC_DIR}" == release* || "${DOC_DIR}" == develop* ]]; then
+    REMOTE_DIR=''
+  else
+    #REMOTE_DIR=branches/${DOC_DIR}
+    REMOTE_DIR=${VERSION}-${DOC_DIR}
+    BRANCH=yes
+  fi
+  decho "SUBDIR=${REMOTE_DIR}"
+}
+
+# parameters that can be passed to this script (as environment variables)
+DEBUG=${DEBUG:-no}
+DEPLOY_TO_STG=${DEPLOY_TO_STG:-no}
+DEPLOY_TO_DOCS=${DEPLOY_TO_DOCS:-no}
+DOCS_SERVER1=${DOCS_SERVER1:-docs1.cask.co}
```

Comment:
no defaults for destination hosts... these should need to be explicitly set (from the build), otherwise fail.